smartsim/_core/entrypoints/redis.py

```diff
@@ -113,28 +112,18 @@ def main(args: argparse.Namespace) -> int:
         *build_bind_args(src_addr, *bind_addrs),
     ]
 
-    # Prevent redirection of stdout and stderr
-    with (
-        open(shard_data.name + ".out", "w", encoding="utf-8")
-        if args.redirect_output
-        else sys.stdout
-    ) as sys.stdout, (
-        open(shard_data.name + ".err", "w", encoding="utf-8")
-        if args.redirect_output
-        else sys.stderr
-    ) as sys.stderr:
-        print_summary(cmd, args.ifname, shard_data)
-
-        try:
-            process = psutil.Popen(cmd, stdout=PIPE, stderr=STDOUT)
-            DBPID = process.pid
-
-            for line in iter(process.stdout.readline, b""):
-                print(line.decode("utf-8").rstrip(), flush=True)
-        except Exception as e:
-            cleanup()
-            raise SSInternalError("Database process starter raised an exception") from e
-        return 0
+    print_summary(cmd, args.ifname, shard_data)
+
+    try:
+        process = psutil.Popen(cmd, stdout=PIPE, stderr=STDOUT)
+        DBPID = process.pid
+
+        for line in iter(process.stdout.readline, b""):
+            print(line.decode("utf-8").rstrip(), flush=True)
+    except Exception as e:
+        cleanup()
+        raise SSInternalError("Database process starter raised an exception") from e
```

Comment:
Do we really need to raise an exception on the way out of an entrypoint? Can we log & return a non-zero?